### PR TITLE
[CI]: run packaging if ^go.mod changes

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

Add packaging stage when a ChangeRequest touches the `^go.mod` files.

## Why is it important?

Early detection of possible issues with the packaging.

## Issues

Closes https://github.com/elastic/beats/issues/21630

## Implementation details

https://beats-ci.elastic.co/job/Beats/job/packaging/ got a pre-validation stage that only gets trigger if:
- GitHub comment event
- Upstream event
- Manual UI event

Git changeset validation would require a worker, so in order to add that particular overhead we use parenstream-downstream relationship rather than the approach of dowsntream subscribed to the parentstream events.

## Tests

https://github.com/elastic/beats/commit/bd1dd763a5688266dfaf255b81f6163ee6f4cd0a

caused 
image

GH checks

image

[build](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fpackaging/detail/PR-22383/6/pipeline/)

### CODE REVIEW

**Review by mdelapenya — APPROVED** [review]

**Comment by mdelapenya** [comment]
I wonder if we want to run the `/packaging` command with any PR 🤔

**Comment by kuisathaverat** [comment]
generate all packages seems a little excessive, run it only for intel and Docker would be enough, but right now it is not possible to select what to build until we merged https://github.com/elastic/beats/pull/22178 that is blocked by the GCP plugin issue.

**Review by cachedout — APPROVED** [review]

**Review by kuisathaverat — APPROVED** [review]

**Pull request merged**
